### ISSUE TITLE
Dragging files (textures, ...) is extremely slow to their respective fields in the Inspector.

### ISSUE THREAD

**Issue by viksl** [godotengine/godot#78471]
### Godot version

4.1 b2

### System information

Windows 10, Nvidia GTX 1660Ti, Vulkan

### Issue description

When you drag a texture to albedo or normal or anything for the standard material from the FileSystem to Inspector, or an HDRI texture to Panorama sky material and so on it's extremely slow depending on the texture size.

Extreme example is an HDRI texture which has tens or hundreds of megabytes which is not a great test case for actual games but it shows the problem well: https://polyhaven.com/a/little_paris_under_tower

Real test cases are mesh textures a 2k or 4k texture which are not that uncommon for 3D are slow to drag too.

You have to drag the file, wait seconds while holding a button waiting if it actually works or not and then if the file appears under your mouse cursor you can then drag it to the appropriate place. If you release the mouse anytime before it shows the drag which can take several seconds then it won't work.

I assume godot is loading the file in some way or so when you drag it to figure out what it is or something? Would it be possible to cache somewhere what the files are so the editor knows where they can and can't go and then do the final load/processing after it's been dragged, please?
I understand that for smaller files it's not an issue but 2k or 4k textures are nothing unusual for 3D games and it's taking forever to texture the models and so on.

### Steps to reproduce

Download the link above, add a PanoramaSky and drag the EXR file into the panorama's texture field.
(I understand the EXR file is huge but that's the purpose of this report, to make sure no one misses how it works, the same issues happens with any regular texture, I assume not just textures but other file types as well)

### Minimal reproduction project

I'll add one with the texture, I have to leave right now but wanted to log this in.

**Comment by Calinou** [comment]
> I assume godot is loading the file in some way or so when you drag it to figure out what it is or something?

When the cursor moves to the 3D viewport, the texture has to be loaded so it can be previewed on materials you drag the texture onto.

**Comment by viksl** [comment]
@Calinou yeah I do make dummy materials but I'm working what I have from various sources ;).
ORMMaterials would certainly be nice, I'll check if there are some tools to compile the textures together.
Even than it is just slow, with a 4k texture triggering the drag functionality takes seconds, I don't have the high end computer but it's also not a complete low end either, which makes it difficult to drag anything anywhere and even when you drag it keeps "lagging" for loss of better words while you drag it around - what is happening during the drag if the texture is already loaded at this point?
Btw I don't move the texture to the 3D viewport at all, below is my layout, I only drag it form the FileSystem to the Inspector straight:
![layout](https://github.com/godotengine/godot/assets/33526847/3df8e91b-6a98-443e-bc1a-9d6139366d87)

Anyway, one thing is how slow it is to initiate the drag which is due to some processing, the second issues is that dragging itself is incredibly slow and freezes for moments all the time too.

I haven't tried it over the 3D Viewport yet :D.

I wonder if the processing could start after the file enters the 3D Viewport only instead of straight away and when it's assigned to a material slot instead of so early?

**Comment by Zireael07** [comment]
1) Can't we defer stuff (whatever it is, not just VRAM compression) AFTER drop is done?
2) Have a switch off for the 3d viewport behavior for some textures (maybe a case by case setting, maybe per size cutoff?)

**Comment by quackmarx** [comment]
I would like to re-open this discussion as the issue persists and at least in my experience is a fair annoyance whenever you deviate from using materials with packaged textures in gltf.

In my eyes, Godot supports alternative workflows because there can be many reasons not to use embedded textures; prototyping, fast texture iteration (eg. not having to re-package through another software to update a texture), custom materials with custom shaders, using 3rd party assets that you don't have control over without having to add the extra and definitely more cumbersome step of streamlining them in a DCC. Supporting these workflows but having the editor hang up when using them is in my opinion not ideal. After all, dragging and dropping textures to a custom material is an expected thing for a user to do in most engines these days.

I'm uncertain about the technical fix required, but I personally think that reducing hang ups in the editor is more important than a live preview in the viewport. Although hopefully there is a possible solution that encompasses both. This is perhaps a dumb question, but is it possible to do these things upon importing the asset so they are already cached during the usual wait-time?

**Comment by Calinou** [comment]
VRAM compression should be greatly sped up with https://github.com/godotengine/godot/pull/95915 in 4.4, so this should be much less noticeable now.

**Comment by that404nerd** [comment]
Actually, it still happens in godot 4.6.2 stable. Dragging PBR files into their respective fields takes like 5-6 sometimes more than 10 seconds. Please look into this. Thanks!

**Comment by Calinou** [comment]
What resolution are the textures you're drag-and-dropping? VRAM compression for BPTC can still be made faster thanks to https://github.com/godotengine/godot/pull/116049, but other formats (S3TC/~~ETC2/ASTC~~) are already using fast GPU-based encoders. The default format used on desktop is S3TC if you haven't changed any VRAM compression settings.

I think the only alternative we have is to no longer show the actual texture when dragging a texture over a mesh, but show a colored preview using a plain color (or pattern) instead and replace it with the actual texture when the drag operation is complete.

**Comment by BlueCube3310** [comment]
> What resolution are the textures you're drag-and-dropping? VRAM compression for BPTC can still be made faster thanks to [#116049](https://github.com/godotengine/godot/pull/116049), but other formats (S3TC/ETC2/ASTC) are already using fast GPU-based encoders. The default format used on desktop is S3TC if you haven't changed any VRAM compression settings.
> 
> I think the only alternative we have is to no longer show the actual texture when dragging a texture over a mesh, but show a colored preview using a plain color (or pattern) instead and replace it with the actual texture when the drag operation is complete.

ETC2 and ASTC are still CPU-based

**Comment by BlueCube3310** [comment]
It looks like dragging a resource over controls keeps reloading it every frame, which is excruciatingly slow when dealing with the default compression mode (lossless)
